requirements-dev.in

```diff
@@ -5,7 +5,7 @@
 # jsonschema==4.21.1
 coverage==7.2.7
 # setuptools==71.1.0;python_version>="3.12"
-# packaging==24.1;python_version>="3.12"  # Requirement for setuptools>=71
+packaging==24.1;python_version>="3.12"  # Requirement for setuptools>=71
```

Comment:
The comment made me think that it's a transitive dependency, and I was about to tell you to use `-c` instead. But then I saw that it's actually direct, in which case, it's best to update the comment.
One reason is that the relationship with `setuptools` is not explained and unclear. Things like `setuptools` or `pip` that can't really have deps in the metadata, vendor other libs, sometimes in a patched form.

They should not be relied upon by other projects. Instead, always define proper dependencies and use those.

So I'd say that the environment marker should be dropped and dependency made unconditional. And the comment should be updated to just say where it's used/imported.

```suggestion
packaging==24.1  # Used in test_version.py
```

Additionally, I recommend not having that `==` in `.in` files but generating `.txt` from those. So `.in` would have unrestricted direct deps, but `.txt` would have everything locked. If there's known broken (transitive) deps to exclude, I like tracking them in a dedicated `.in` file wired into this one via `-c`: https://github.com/ansible/awx-plugins/blob/53ebc59/dependencies/direct/py.in#L1 / https://github.com/ansible/awx-plugins/blob/53ebc59/dependencies/direct/py-constraints.in#L17.

Reply:
The dependency specification along with the comment is actually copied verbatim from upstream (see comment in line one of that file). To make future drift less likely, I would leave it as is, but don't fell strongly about it.

The `.in` file will be gone once we migrate from pip-tools to uv. This PR is one of several prerequisites to prepare for that change.

Reply:
If you mix files from upstream with local changes, I recommend vendoring them separately and including into yours. This allows you to sync the contents by copying blindly.

Reply:
> If you mix files from upstream with local changes, I recommend vendoring them separately and including into yours. This allows you to sync the contents by copying blindly.

Hm, I will have to think about that. Sounds reasonable for sure.

Reply:
Opened separate issue: #1251

Reply:
How would that work? For the purpose of this PR, I'll redo the commenting out of the packaging requirement from upstream and put a new, unconditional entry in the bottom of that file to express our direct use of packaging in tests.